From 9da59a738d9fc80ebd4799ed67f1e258cf54a2f6 Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexander.deucher@amd.com>
Date: Wed, 30 Mar 2016 11:43:16 -0400
Subject: [PATCH 0969/1110] drm/amdgpu: clean up dal checks in set_ip_blocks
 functions

There's no need to check amdgpu_dal.  It's already checked
in amdgpu_device_has_dal_support.

Reviewed-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/cik.c | 4 ++--
 drivers/gpu/drm/amd/amdgpu/vi.c  | 8 ++++----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/cik.c b/drivers/gpu/drm/amd/amdgpu/cik.c
index 7efe693..14a1411 100644
--- a/drivers/gpu/drm/amd/amdgpu/cik.c
+++ b/drivers/gpu/drm/amd/amdgpu/cik.c
@@ -2112,7 +2112,7 @@ int cik_set_ip_blocks(struct amdgpu_device *adev)
 	switch (adev->asic_type) {
 	case CHIP_BONAIRE:
 #if defined(CONFIG_DRM_AMD_DAL_DCE8_0)
-		if (amdgpu_dal && amdgpu_device_has_dal_support(adev)) {
+		if (amdgpu_device_has_dal_support(adev)) {
 			adev->ip_blocks = bonaire_ip_blocks_dal;
 			adev->num_ip_blocks = ARRAY_SIZE(bonaire_ip_blocks_dal);
 		} else {
@@ -2126,7 +2126,7 @@ int cik_set_ip_blocks(struct amdgpu_device *adev)
 		break;
 	case CHIP_HAWAII:
 #if defined(CONFIG_DRM_AMD_DAL_DCE8_0)
-		if (amdgpu_dal && amdgpu_device_has_dal_support(adev)) {
+		if (amdgpu_device_has_dal_support(adev)) {
 			adev->ip_blocks = hawaii_ip_blocks_dal;
 			adev->num_ip_blocks = ARRAY_SIZE(hawaii_ip_blocks_dal);
 		} else {
diff --git a/drivers/gpu/drm/amd/amdgpu/vi.c b/drivers/gpu/drm/amd/amdgpu/vi.c
index 8d876a7..b5602ac 100644
--- a/drivers/gpu/drm/amd/amdgpu/vi.c
+++ b/drivers/gpu/drm/amd/amdgpu/vi.c
@@ -1359,7 +1359,7 @@ int vi_set_ip_blocks(struct amdgpu_device *adev)
 		break;
 	case CHIP_FIJI:
 #if defined(CONFIG_DRM_AMD_DAL)
-		if (amdgpu_dal && amdgpu_device_has_dal_support(adev)) {
+		if (amdgpu_device_has_dal_support(adev)) {
 			adev->ip_blocks = fiji_ip_blocks_dal;
 			adev->num_ip_blocks = ARRAY_SIZE(fiji_ip_blocks_dal);
 		} else {
@@ -1373,7 +1373,7 @@ int vi_set_ip_blocks(struct amdgpu_device *adev)
 		break;
 	case CHIP_TONGA:
 #if defined(CONFIG_DRM_AMD_DAL)
-		if (amdgpu_dal && amdgpu_device_has_dal_support(adev)) {
+		if (amdgpu_device_has_dal_support(adev)) {
 			adev->ip_blocks = tonga_ip_blocks_dal;
 			adev->num_ip_blocks = ARRAY_SIZE(tonga_ip_blocks_dal);
 		} else {
@@ -1388,7 +1388,7 @@ int vi_set_ip_blocks(struct amdgpu_device *adev)
 	case CHIP_POLARIS11:
 	case CHIP_POLARIS10:
 #if defined(CONFIG_DRM_AMD_DAL)
-		if (amdgpu_dal && amdgpu_device_has_dal_support(adev)) {
+		if (amdgpu_device_has_dal_support(adev)) {
 			adev->ip_blocks = polaris11_ip_blocks_dal;
 			adev->num_ip_blocks = ARRAY_SIZE(polaris11_ip_blocks_dal);
 		} else {
@@ -1403,7 +1403,7 @@ int vi_set_ip_blocks(struct amdgpu_device *adev)
 	case CHIP_CARRIZO:
 	case CHIP_STONEY:
 #if defined(CONFIG_DRM_AMD_DAL)
-		if (amdgpu_dal && amdgpu_device_has_dal_support(adev)) {
+		if (amdgpu_device_has_dal_support(adev)) {
 			adev->ip_blocks = cz_ip_blocks_dal;
 			adev->num_ip_blocks = ARRAY_SIZE(cz_ip_blocks_dal);
 		} else {
-- 
2.7.4

